### PR TITLE
Fix volume initialize error check

### LINKED ISSUE #11725: Volume initialize() code invalid
**Problem**

The code in https://github.com/docker/docker/blob/master/volumes/volume.go#L93 is incorrect.  It only hits the inner side of the `if` block if the error is not nil and it's an is not exist error.  Any other error type is ignored and the code just goes along.

**Solution**

Fix the logic to only mkdirall when the error is not exist and return the error for any other error type.

### PR DESCRIPTION
**Pull request by runcom**
Fixes #11725 and adds regression tests

Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

File: volumes/volume.go [hunk 1]
```diff
@@ -90,8 +90,13 @@ func (v *Volume) initialize() error {
 	v.lock.Lock()
 	defer v.lock.Unlock()
 
-	if _, err := os.Stat(v.Path); err != nil && os.IsNotExist(err) {
-		if err := os.MkdirAll(v.Path, 0755); err != nil {
+	if _, err := os.Stat(v.Path); err != nil {
+		switch {
```
Comment (cpuguy83) [inline volumes/volume.go]: Maybe instead of this switch, do:
```golang
if !os.IsNotExist(err) {
    return err
}

// mkdir code here
```

Comment (runcom) [inline volumes/volume.go]: I thought to use `switch` in order to chain future error messages in there and not with many `if`s. Alright btw :+1:

**Comment by runcom** [comment]
Fixed @cpuguy83 (seems like TestPullVerified is failing on windows though)

**Comment by LK4D4** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**